### DISCUSSION TITLE
api alternative fetch method

### DISCUSSION THREAD

**Discussion by s-b-repo** [KRTirtho/spotube#1994] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Pain

api access could get denied, blocked or rate limited

### Suggested solution

we could use a web scraping tool get the audio directly from the page without rendering video like blocking or ignoring or dropping the video connection then get audio from youtube

### Useful resources

course
https://www.youtube.com/watch?v=XVv6mJpFOb0&pp=ygUUd2ViIHNjcmFwaW5nICBjb3Vyc2U%3D

### Self grab

- [X] I'm ready to work on this issue!

**Comment by s-b-repo** [comment]
also we can use a user-agent spoofer to avoid detection while scraping like [chamelion ](https://github.com/sereneblue/chameleon)

**Comment by KRTirtho** [comment]
This is not how it works here. Spotube uses a library [youtube_explode_dart](https://pub.dev/packages/youtube_explode_dart). And we only use the audio stream of the video. Not the whole video
